unit_test/tstGridOperator2d.hpp

```diff
@@ -0,0 +1,196 @@
+
+        // Set up the local dependency to be the identity.
+        baz( i, j ) = { { 1.0, 0.0 }, { 0.0, 1.0 } };
+
+        // Assign foo_out - build a point-wise expression to do this.
+        auto baz_t_foo_in_t_2 =
+            baz( i, j ) * ( foo_in( i, j ) + foo_in( i, j ) );
+        for ( int d = 0; d < 2; ++d )
+            foo_out_access( i, j, d ) += baz_t_foo_in_t_2( d ) + i + j;
+
+        // Assign the bar_out - use mixture of field dimension indices
+        // or direct ijk access.
+        bar_out_access( i, j, 0 ) += bar_in( i, j, 0 ) + bar_in( i, j ) + i + j;
+    }
+};
+
+//---------------------------------------------------------------------------//
+void gatherScatterTest()
+{
+    // Global bounding box.
+    double cell_size = 0.23;
+    std::array<int, 2> global_num_cell = { 43, 22 };
+    std::array<double, 2> global_low_corner = { 1.2, 2.3 };
+    std::array<double, 2> global_high_corner = {
+        global_low_corner[0] + cell_size * global_num_cell[0],
+        global_low_corner[1] + cell_size * global_num_cell[1] };
+    Kokkos::Array<bool, 2> periodic = { false, false };
+    int comm_size = -1;
+    MPI_Comm_size( MPI_COMM_WORLD, &comm_size );
+    std::array<int, 2> ranks_per_dim = { comm_size, 1 };
+
+    // Get inputs for mesh.
+    Kokkos::Array<double, 4> global_box = {
+        global_low_corner[0], global_low_corner[1], global_high_corner[0],
+        global_high_corner[1] };
+    int minimum_halo_size = 0;
+
+    // Make mesh and field manager.
+    auto fm = createUniformCartesianMesh(
+        TEST_MEMSPACE{}, cell_size, global_box, periodic, minimum_halo_size,
+        MPI_COMM_WORLD, ranks_per_dim );
+
+    // Make an operator.
+    using gather_deps =
+        GatherDependencies<FieldLayout<FieldLocation::Cell, FooIn>,
+                           FieldLayout<FieldLocation::Cell, BarIn>>;
+    using scatter_deps =
+        ScatterDependencies<FieldLayout<FieldLocation::Cell, FooOut>,
+                            FieldLayout<FieldLocation::Cell, BarOut>>;
+    using local_deps = LocalDependencies<FieldLayout<FieldLocation::Cell, Baz>>;
+    auto grid_op = createGridOperator( fm->mesh(), gather_deps(),
+                                       scatter_deps(), local_deps() );
+
+    // Setup the field manager.
+    grid_op->setup( *fm );
+
+    // Initialize gather fields.
+    Kokkos::deep_copy( fm->view( FieldLocation::Cell(), FooIn() ), 2.0 );
+    Kokkos::deep_copy( fm->view( FieldLocation::Cell(), BarIn() ), 3.0 );
+
+    // Initialiize scatter fields to wrong data to make sure they get reset to
+    // zero and then overwritten with the data we assign in the operator.
+    Kokkos::deep_copy( fm->view( FieldLocation::Cell(), FooOut() ), -1.1 );
+    Kokkos::deep_copy( fm->view( FieldLocation::Cell(), BarOut() ), -2.2 );
+
+    // Apply the grid operator. Use a tag.
+    GridFunc grid_func;
+    grid_op->apply( FieldLocation::Cell(), TEST_EXECSPACE(), *fm,
```

Comment:
So particles will be tested here too once they support 2D?

Reply:
Yes. I started down this path but realized it would be a bit of work and more pull requests would be needed.